### ISSUE TITLE
const char* function input is returned to heap too early

### ISSUE THREAD

**Issue by ryannevell** [jasonroelofs/rice#97]
I am trying to wrap a function that takes a `const char*` input argument. Inside of this function, it is allocating its own resources using `new`, howerver, sometimes it seems that `new` returns a pointer to the same region of memory that the input argument was using.

```
#include "rice/Class.hpp"
#include "rice/String.hpp"
#include <iostream>

using namespace Rice;

void test_hello(const char *str)
{
  char *tmp = new char[23];
  std::cout << "str = " << (void*)str << ", tmp = " << (void*)tmp << std::endl;
}

extern "C"
void Init_test()
{
  Class rb_cTest =
    define_class("Test")
      .define_method("hello", &test_hello);
}
```

```
irb(main):001:0> GC.disable
irb(main):002:0> require './test'
irb(main):003:0> Test.new.hello('X' * 23)
str = 0x7f90a2901c70, tmp = 0x7f90a2901c70
```

This likely points to some temporary copy of the Ruby string being freed back to the heap prior to the function call.

**Comment by jasonroelofs** [comment]
`from_ruby` for `const char *` has been a regular point of problems, though I am not sure if that's the nature of the C++ <=> Ruby difficulty of figuring out what exactly this is (e.g. a pointer to a single `char` or a null-terminated string), my admittedly weak knowledge of C++ memory handling, or a bit of both.

You say this sometimes looks weird. Are you experiencing other issues that led you to check out the raw pointer locations?

The actual conversion happens here: https://github.com/jasonroelofs/rice/blob/master/rice/to_from_ruby.ipp#L264. If you know of a better, more memory safe way to handle this I'd love suggestions.

**Comment by jasonroelofs** [comment]
Should now be fixed with #98.
